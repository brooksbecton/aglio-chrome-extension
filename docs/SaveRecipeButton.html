<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: SaveRecipeButton</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: SaveRecipeButton</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>SaveRecipeButton<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">UI for saving the user's current page</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="SaveRecipeButton"><span class="type-signature"></span>new SaveRecipeButton<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Initializes State
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="SaveRecipeButton_index.js.html">SaveRecipeButton/index.js</a>, <a href="SaveRecipeButton_index.js.html#line7">line 7</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="displayError"><span class="type-signature"></span>displayError<span class="type-signature"></span></h4>




<div class="description">
    Sets the state of 'saveError' to a supplied error message
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="SaveRecipeButton_index.js.html">SaveRecipeButton/index.js</a>, <a href="SaveRecipeButton_index.js.html#line27">line 27</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="saveRecipe"><span class="type-signature">(async) </span>saveRecipe<span class="type-signature"></span></h4>




<div class="description">
    Driver function to save a recipe
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="SaveRecipeButton_index.js.html">SaveRecipeButton/index.js</a>, <a href="SaveRecipeButton_index.js.html#line35">line 35</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="AllRecipesScraper.html">AllRecipesScraper</a></li><li><a href="App.html">App</a></li><li><a href="BaseScrape.html">BaseScrape</a></li><li><a href="SaveRecipeButton.html">SaveRecipeButton</a></li><li><a href="ScraperFactory.html">ScraperFactory</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Thu May 17 2018 13:42:36 GMT-0500 (CDT)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>