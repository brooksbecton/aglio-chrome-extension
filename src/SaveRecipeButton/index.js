/*global chrome*/

import React, { Component } from "react";
import PropTypes from "prop-types";

import ScraperFactory from "./../lib/scrapers/ScraperFactory";

/** Saves the users current page content */
class SaveRecipeButton extends Component {
  static propTypes = {
    handleSave: PropTypes.func
  };

  /**
   * Initializes State
   */
  constructor() {
    super();

    this.state = {
      saveError: ""
    };
  }

  /**
   * Captures the page source of the page the user's active page
   * Uses {@link https://developer.chrome.com/extensions/pageCapture | pageCapture}
   * @returns <Promise> - represents the html of the current page
   */
  capturePage = () => {
    return new Promise((resolve, reject) => {
      chrome.tabs.executeScript(
        { code: "document.querySelector('body').innerHTML" },
        ({ 0: html }) => {
          resolve(html);
        }
      );
    });
  };

  /**
   * Sets the state of 'saveError' to a supplied error message
   * @param {Error} err
   */
  displayError = err => {
    this.setState({ saveError: err });
  };

  /**
   * Driver function to save a recipe
   * @async
   */
  saveRecipe = async () => {
    let html = await this.capturePage();
  };
  render() {
    return (
      <div>
        <button onClick={() => this.saveRecipe()}>Save Recipe</button>
        <p className="error">
          {this.state.saveError.length > 0 && this.state.saveError}
        </p>
      </div>
    );
  }
}

export default SaveRecipeButton;
