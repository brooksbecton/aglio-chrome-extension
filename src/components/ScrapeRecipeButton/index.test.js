import React from "react";
import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import SaveRecipeButton from "./index";

Enzyme.configure({ adapter: new Adapter() });

describe("SaveRecipeButton", () => {
  it("matches snapshot", () => {
    const tree = shallow(<SaveRecipeButton scrapePageRecipe={() => {}} />);
    expect(tree).toMatchSnapshot();
  });

  it("has a button to click", () => {
    const tree = mount(<SaveRecipeButton scrapePageRecipe={() => {}} />);
    expect(tree.find("button").length).toBe(1);
  });

  it("calls scrapePageRecipe() when button is clicked", () => {
    const tree = mount(<SaveRecipeButton scrapePageRecipe={() => {}} />);
    const mockSaveRecipe = jest.fn();
    tree.instance().scrapePageRecipe = mockSaveRecipe;
    tree.find("button").simulate("click");
    expect(mockSaveRecipe).toHaveBeenCalled();
  });

  it("displays error", () => {
    const errorMessage = "BAM!";
    const tree = shallow(<SaveRecipeButton scrapePageRecipe={() => {}} />);
    tree.setState({ scrapeError: errorMessage });

    expect(tree.find("p").text()).toBe(errorMessage);
  });

  it("displayError() sets 'scrapeError' state", () => {
    const errorMessage = "asdfasdfasdf";
    const tree = shallow(<SaveRecipeButton scrapePageRecipe={() => {}} />);
    tree.instance().displayError(errorMessage);

    expect(tree.state().scrapeError).toBe(errorMessage);
  });
});
