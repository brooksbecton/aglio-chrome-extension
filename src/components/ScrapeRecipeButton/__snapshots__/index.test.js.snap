// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveRecipeButton matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SaveRecipeButton
    scrapePageRecipe={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="saveButton"
          disabled={false}
          onClick={[Function]}
>
          Pull Recipe
</Button>,
        <p
          className="error"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Pull Recipe",
          "className": "saveButton",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Pull Recipe",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "error",
        },
        "ref": null,
        "rendered": false,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="saveButton"
            disabled={false}
            onClick={[Function]}
>
            Pull Recipe
</Button>,
          <p
            className="error"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Pull Recipe",
            "className": "saveButton",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Pull Recipe",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "error",
          },
          "ref": null,
          "rendered": false,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
